### ISSUE TITLE
Support FOP 1.1 [sf#108]

### ISSUE THREAD

**Issue by tomschr** [openSUSE/daps#108]
**Reported by tom_schr on 2012-10-26 17:49 UTC**
This ticket is just for collecting some information about FOP 1.1 and its features. I think, it is worth implementing it in DAPS.
- Disabeling Complex Scripts with <tt>-nocs</tt> 
  [Complex scripts](http://xmlgraphics.apache.org/fop/1.1/complexscripts.html#Disabling+complex+scripts) are a new feature of FOP 1.1. However, as we don't create manuals with BIDI and complex script support needs a lot of memory, I think we can savely disable that feature. Use the <tt>-nocs</tt> option on the command line.
- Support PDF Images
  It seems, we cannot use PDF graphics with FOP directly. We need a special JAR file which supports [PDF Images](http://www.jeremias-maerki.ch/download/fop/pdf-images/). Either we fix our <tt>daps-fop</tt> script and add the JAR to the <tt>ADDITIONAL_JARS</tt> environment variable, or we add it to the xmlgraphics-fop package. See separate Ticket [#179].
- Use FOP Extensions 
  FOP 1.1 supports some nice [FO extensions](http://xmlgraphics.apache.org/fop/1.1/extensions.html) like PDF bookmarks, <tt>fox:external-document</tt>, prepress support and more.
  Probably, these features are better supported upstream. I add them here so we don't forget it.
  See also discussion:
  *\* [Add support for fox:font-feature-settings](https://issues.apache.org/jira/browse/FOP-2256)
  *\* [Add support for fox:complex-scripts property, applicable to block level FOs](https://issues.apache.org/jira/browse/FOP-2257)
- Support (more?) Metadata
  We should investigate, if we could transform more [metadata](http://xmlgraphics.apache.org/fop/1.1/metadata.html) from our DocBook sources into our FO.

@Frank: Probably, the first feature is only interesting for you. After you have implemented it, assign it back to me, please. :-)

**Comment by tomschr** [comment]
**Commented by tom_schr on 2012-11-26 07:58**
- Description has changed:

Diff:

~~~~

--- old
+++ new
@@ -1,13 +1,16 @@
 This ticket is just for collecting some information about FOP 1.1 and its features. I think, it is worth implementing it in DAPS.
 
-* Disabeling complex scripts with <tt>-nocs</tt> 
+* Disabeling Complex Scripts with <tt>-nocs</tt> 
 [Complex scripts](http://xmlgraphics.apache.org/fop/1.1/complexscripts.html#Disabling+complex+scripts) are a new feature of FOP 1.1. However, as we don't create manuals with BIDI and complex script support needs a lot of memory, I think we can savely disable that feature. Use the <tt>-nocs</tt> option on the command line.
 
-* Use FOP extensions 
+* Support PDF Images
+It seems, we cannot use PDF graphics with FOP directly. We need a special JAR file which supports [PDF Images](http://www.jeremias-maerki.ch/download/fop/pdf-images/). Either we fix our <tt>daps-fop</tt> script and add the JAR to the <tt>ADDITIONAL_JARS</tt> environment variable, or we add it to the xmlgraphics-fop package.
+
+* Use FOP Extensions 
 FOP 1.1 supports some nice [FO extensions](http://xmlgraphics.apache.org/fop/1.1/extensions.html) like PDF bookmarks, <tt>fox:external-document</tt>, prepress support and more.
 Probably, these features are better supported upstream. I add them here so we don't forget it.
 
-* Support (more?) metadata
+* Support (more?) Metadata
 We should investigate, if we could transform more [metadata](http://xmlgraphics.apache.org/fop/1.1/metadata.html) from our DocBook sources into our FO.
 
 

~~~~

**Comment by tomschr** [comment]
**Commented by randombauer on 2012-12-06 14:48**
PDF Image support doesn't work with the JAR-file, it produces exception errors or PDFs with no images.
Next test steps:
- check the PDF Image-support JAR implementation for errors (maybe something is missing)
- try the newest xmlgraphics-fop-commons (which might have PDF image support integrated)
- look for a different pdf image support JAR
- use eps instead of pdf

**Comment by tomschr** [comment]
**Commented by randombauer on 2012-12-12 10:01**
PDF Image support works now:
1. Download the JARs from the source above
2. Move 'fop-pdf-images.jar pdfbox.jar jempbox.jar fontbox.jar' to /usr/share/java/
3. Add the JAR names to the 'BASE_JARS' in the fop-script at /usr/bin/fop and to the 'ADDITIONAL_JARS' in daps-fop at daps/libexec/daps-fop

**Comment by tomschr** [comment]
**Updated by fsundermeyer on 2013-01-15 14:24**
- **milestone**: 1.x --> 2.0

**Comment by tomschr** [comment]
**Updated by fsundermeyer on 2013-04-11 21:34**
- **assigned_to**: Frank Sundermeyer --> Thomas Schraitle

**Comment by tomschr** [comment]
**Commented by fsundermeyer on 2013-04-11 21:34**
Added the -nocs option for fop >= 1.1 to libexec/daps-fop with SVN commit [r2933].
Reassingning.

**Comment by tomschr** [comment]
**Commented by tom_schr on 2013-07-01 07:43**
- Description has changed:

Diff:

~~~~

--- old
+++ new
@@ -9,6 +9,10 @@
 * Use FOP Extensions 
 FOP 1.1 supports some nice [FO extensions](http://xmlgraphics.apache.org/fop/1.1/extensions.html) like PDF bookmarks, <tt>fox:external-document</tt>, prepress support and more.
 Probably, these features are better supported upstream. I add them here so we don't forget it.
+See also discussion:
+** [Add support for fox:font-feature-settings](https://issues.apache.org/jira/browse/FOP-2256)
+** [Add support for fox:complex-scripts property, applicable to block level FOs](https://issues.apache.org/jira/browse/FOP-2257)
+
 
 * Support (more?) Metadata
 We should investigate, if we could transform more [metadata](http://xmlgraphics.apache.org/fop/1.1/metadata.html) from our DocBook sources into our FO.

~~~~

**Comment by tomschr** [comment]
**Commented by tom_schr on 2013-07-01 07:56**
Unfortunately, it seems it is not that easy. 

The <tt>-nocs</tt> option is language dependant. Complex scripting is needed for Arabic and CJK, but not Latin. 

How could we do that? Could we use the DC files for that? Or should DAPS do it automatically?

**Comment by tomschr** [comment]
**Commented by tom_schr on 2014-02-04 14:35**
- Description has changed:

Diff:

~~~~

--- old
+++ new
@@ -4,7 +4,7 @@
 [Complex scripts](http://xmlgraphics.apache.org/fop/1.1/complexscripts.html#Disabling+complex+scripts) are a new feature of FOP 1.1. However, as we don't create manuals with BIDI and complex script support needs a lot of memory, I think we can savely disable that feature. Use the <tt>-nocs</tt> option on the command line.
 
 * Support PDF Images
-It seems, we cannot use PDF graphics with FOP directly. We need a special JAR file which supports [PDF Images](http://www.jeremias-maerki.ch/download/fop/pdf-images/). Either we fix our <tt>daps-fop</tt> script and add the JAR to the <tt>ADDITIONAL_JARS</tt> environment variable, or we add it to the xmlgraphics-fop package.
+It seems, we cannot use PDF graphics with FOP directly. We need a special JAR file which supports [PDF Images](http://www.jeremias-maerki.ch/download/fop/pdf-images/). Either we fix our <tt>daps-fop</tt> script and add the JAR to the <tt>ADDITIONAL_JARS</tt> environment variable, or we add it to the xmlgraphics-fop package. See separate Ticket [#179].
 
 * Use FOP Extensions 
 FOP 1.1 supports some nice [FO extensions](http://xmlgraphics.apache.org/fop/1.1/extensions.html) like PDF bookmarks, <tt>fox:external-document</tt>, prepress support and more.

~~~~

**Comment by tomschr** [comment]
**Commented by fsundermeyer on 2014-10-23 13:27**
I vote for not disabling complex scripts - I wasn't able to see any differences between a "fop" and a "fop -nocs" run on my test machine. memory consumption was not higher without -nocs.

**Comment by ghost** [comment]
@tomschr Is this fixed? I think we have all been using FOP 1.1 for some time now.

Of the original items this bug was opened for, I see discussion on 2 of 3:
- Disable Complex Scripts with -nocs = _off the table_
- Support PDF Images = _done_
- Use FOP Extensions = _no discussion_, though this is probably a stylesheet topic.

**Comment by fsundermeyer** [comment]
Looks like this can be closed....
